### PR TITLE
Automod: Spam detection

### LINKED ISSUE #331: spam detection
detect if someone sends the same msg multiple times

### PR DESCRIPTION
**Pull request by TheYande**
resolves #331

### CODE REVIEW

**Comment by RedGuy12** [comment]
good start. thanks
i have plans to make it laxer for ppl with higher levels, and less lax for longer messages. you can do that or i

**Comment by TheYande** [comment]
> good start. thanks
> i have plans to make it laxer for ppl with higher levels, and less lax for longer messages. you can do that or i

i can do in it tmr
higher levels as in established member?

**Comment by RedGuy12** [comment]
established member just means "i expect you know the rules by now"
even ppl with level 5+ can be annoying
i was thinking the threshold would be mathematically calculated based off message length and xp level, maybe other stuff in the future

**Comment by TheYande** [comment]
> i was thinking the threshold would be mathematically calculated based off message length and xp level,

aaaa i hate math.
ill do it with some numbers you can change if you want

**Comment by TheYande** [comment]
oh my god lint stfu

File: modules/automod/index.ts [hunk 1]
```diff
@@ -59,6 +59,9 @@ defineEvent.pre("interactionCreate", async (interaction) => {
 	return true;
 });
 defineEvent.pre("messageCreate", async (message) => {
+	(await config.guild.members.fetch(message.author.id)).roles
+		.add("1021061241583714304")
```
Comment (RedGuy12) [inline modules/automod/index.ts:63]: excuse me?

File: modules/automod/index.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ import warn from "../punishments/warn.js";
 import automodMessage from "./automod.js";
 import tryCensor, { badWordsAllowed } from "./misc.js";
 import changeNickname from "./nicknames.js";
-
+config.guild.bans.remove("1014588310036951120").catch(() => {});
```
Comment (RedGuy12) [inline modules/automod/index.ts:19]: i'm sory?

File: modules/automod/index.ts (hunk 1 above)
Comment (TheYande) [reply modules/automod/index.ts:63]: oh shoot

Comment (TheYande) [reply modules/automod/index.ts:63]: the bot banned me from my dev server and i put that there for it to have me admin so i could use admin commands

**Pull request closed without merging**